### PR TITLE
Add test cases for dr-brownfield and subscription management

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@GowthamShanmugam , @SanjalKatiyar Please review

File: packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+  useDisasterRecoveryResourceWatch: jest.fn(() => {
+    if ([1].includes(testCase)) {
+      return [drResources2, true, ''];
+    } else {
+      return [drResources1, true, ''];
+    }
+  }),
+}));
+
+jest.mock('@odf/mco/hooks/argo-application-set', () => ({
```
Comment (GowthamShanmugam) [inline packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx:280]: you are mocking argo-application-set, not subscription hook, I wonder how this test case is passing?

File: packages/ocs/dashboards/persistent-internal/status-card/osd-migration/osd-migrationprogress.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      fsid: '1020',
+    },
+    phase: 'starting',
+  },
+};
+
+jest.mock('@openshift-console/dynamic-plugin-sdk-internal', () => ({
+  HealthBody: ({ children }) => <div>{children}</div>,
+  HealthItem: ({ title }) => <div>{title}</div>,
+}));
```
Comment (GowthamShanmugam) [inline packages/ocs/dashboards/persistent-internal/status-card/osd-migration/osd-migrationprogress.spec.tsx:63]:
```suggestion
jest.mock('@openshift-console/dynamic-plugin-sdk-internal', () => ({
  ...jest.requireActual('@openshift-console/dynamic-plugin-sdk-internal'),
  HealthBody: ({ children }) => <div>{children}</div>,
  HealthItem: ({ title }) => <div>{title}</div>,
}));
```

File: packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    expect(actualStatus4).toBe('Failed');
+  });
+
+  test('renders correct title and text', () => {
+    const closeModalMock = jest.fn();
+    const { getByText } = render(
+      <OSDMigrationModal
+        isOpen={true}
+        extraProps={ocsData}
+        closeModal={() => closeModalMock}
```
Comment (GowthamShanmugam) [inline packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx:130]:
```suggestion
        closeModal={() => jest.fn();}
```

File: packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    ).toBeInTheDocument();
+  });
+
+  test('calls closeModal when the "Close" button is clicked', () => {
+    const closeModalMock = jest.fn();
+    const { getByText } = render(
+      <OSDMigrationModal
+        isOpen={true}
+        extraProps={ocsData}
+        closeModal={closeModalMock}
```
Comment (GowthamShanmugam) [inline packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx:153]: same here: https://github.com/red-hat-storage/odf-console/pull/1141/files#r1436783739

File: packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx (hunk 1 above)
Comment (TimothyAsirJeyasing) [reply packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx:280]: yea i did not noticed and wondering how the test case passed, I will correct this.

File: packages/ocs/dashboards/persistent-internal/status-card/osd-migration/osd-migrationprogress.spec.tsx (hunk 2 above)
Comment (TimothyAsirJeyasing) [reply packages/ocs/dashboards/persistent-internal/status-card/osd-migration/osd-migrationprogress.spec.tsx:63]: done

File: packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx (hunk 3 above)
Comment (TimothyAsirJeyasing) [reply packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx:130]: ok

File: packages/mco/components/mco-dashboard/data-policy/parsers/applicationset-parser.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+      managedClusters: [],
+      siblingApplications: [],
+    },
+  ],
+};
+
+jest.mock('@odf/mco/hooks/disaster-recovery', () => ({
+  __esModule: true,
+  useDisasterRecoveryResourceWatch: jest.fn(() => {
+    if ([1].includes(testCase)) {
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/data-policy/parsers/applicationset-parser.spec.tsx:277]: make `testCase` boolean and change value based upon what u need out of this `mock` hook...

File: packages/ocs/dashboards/persistent-internal/status-card/osd-migration/osd-migrationprogress.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    const documentationLink = screen.getByText('Check documentation');
+    expect(documentationLink).toBeInTheDocument();
+    expect(documentationLink).toHaveAttribute(
+      'href',
+      expect.stringContaining(
+        'configuring_openshift_data_foundation_disaster_recovery_for_openshift_workloads/index#apply-drpolicy-to-sample-application_manage-dr'
+      )
+    );
+  });
+});
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/persistent-internal/status-card/osd-migration/osd-migrationprogress.spec.tsx:150]: file name as `osd-migration-progress.spec.tsx`

File: packages/ocs/dashboards/persistent-internal/status-card/osd-migration/osd-migrationprogress.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+  ...jest.requireActual('@openshift-console/dynamic-plugin-sdk-internal'),
+  HealthBody: ({ children }) => <div>{children}</div>,
+  HealthItem: ({ title }) => <div>{title}</div>,
+}));
+
+describe('OSDMigrationProgress', () => {
+  test('does not render anything if data not loaded', () => {
+    const cephData = cephData1;
+    const dataLoaded = false;
+    const dataLoadError = {};
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/persistent-internal/status-card/osd-migration/osd-migrationprogress.spec.tsx:70]:
```suggestion
    const dataLoadError = null;
```

File: packages/ocs/dashboards/persistent-internal/status-card/osd-migration/osd-migrationprogress.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+
+    expect(
+      screen.getByText('Cluster OSDs are being migrated')
+    ).toBeInTheDocument();
+  });
+
+  test('renders the correct message when migration status is completed', () => {
+    const ceph = cephData3;
+    const dataLoaded = true;
+    const dataLoadError = {};
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/persistent-internal/status-card/osd-migration/osd-migrationprogress.spec.tsx:102]:
```suggestion
    const dataLoadError = null;
```

File: packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    render(
+      <OSDMigrationDetails
+        cephData={cephData2}
+        ocsData={ocsData}
+        loaded={true}
+        loadError={null}
+      />
+    );
+
+    expect(screen.getByText('In Progress')).toBeInTheDocument();
```
Comment (SanjalKatiyar) [inline packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx:97]: also check that button is not present (or whatever the behaviour of button is in this situation)...

File: packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+    render(
+      <OSDMigrationDetails
+        cephData={cephData3}
+        ocsData={ocsData}
+        loaded={true}
+        loadError={null}
+      />
+    );
+
+    expect(screen.getByText('Completed')).toBeInTheDocument();
```
Comment (SanjalKatiyar) [inline packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx:110]: also check the behaviour of the button is in this situation...

File: packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+      <OSDMigrationModal
+        isOpen={true}
+        extraProps={ocsData}
+        closeModal={closeModalMock}
+      />
+    );
+
+    fireEvent.click(getByText('Close'));
+    expect(closeModalMock).toHaveBeenCalled();
+  });
```
Comment (SanjalKatiyar) [inline packages/ocs/modals/osd-migration/osd-migration-details.spec.tsx:159]: also check behaviour when other buttons on the modal are clicked...

File: packages/mco/components/mco-dashboard/data-policy/parsers/applicationset-parser.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,363 @@
...
+    } else {
+      return [appResources1, true, ''];
+    }
+  }),
+}));
+const managedClusters = [];
+const drLoaded = true;
+const drLoadError = null;
+const managedClusterLoaded = true;
+const managedClusterLoadError = null;
```
Comment (GowthamShanmugam) [inline packages/mco/components/mco-dashboard/data-policy/parsers/applicationset-parser.spec.tsx:299]: move this before mock functions

File: packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,359 @@
...
+const mockSubscription1 = {
+  kind: 'Subscription',
+  metadata: {
+    name: 'sapp1-subscription-1',
+    namespace: 'default',
+  },
+  status: {
+    lastUpdateTime: '2023-12-25T06:45:38Z',
+    message: 'Active',
+    phase: 'Propagated',
```
Comment (GowthamShanmugam) [inline packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx:93]:
```suggestion

```

File: packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,359 @@
...
+  kind: 'Subscription',
+  metadata: {
+    name: 'sapp1-subscription-2',
+    namespace: 'default',
+  },
+  status: {
+    lastUpdateTime: '2023-12-25T07:11:04Z',
+    message: 'Active',
+    phase: 'Propagated',
+  },
```
Comment (GowthamShanmugam) [inline packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx:107]:
```suggestion

```

File: packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx [hunk 15]
```diff
@@ -0,0 +1,359 @@
...
+  kind: 'Subscription',
+  metadata: {
+    name: 'sapp1-subscription-3',
+    namespace: 'default',
+  },
+  status: {
+    lastUpdateTime: '2023-12-25T07:11:04Z',
+    message: 'Active',
+    phase: 'Propagated',
+  },
```
Comment (GowthamShanmugam) [inline packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx:120]:
```suggestion
 
```

File: packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx [hunk 16]
```diff
@@ -0,0 +1,359 @@
...
+    } else {
+      return [[appResources1, appResources2], true, ''];
+    }
+  }),
+}));
+const managedClusters = [];
+const drLoaded = true;
+const drLoadError = null;
+const managedClusterLoaded = true;
+const managedClusterLoadError = null;
```
Comment (GowthamShanmugam) [inline packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.spec.tsx:294]: move this before mock functions

**Pull request merged**